diff --git a/vim-tmux-navigator.tmux b/vim-tmux-navigator.tmux
index 0aeac55..3af48bc 100755
--- a/vim-tmux-navigator.tmux
+++ b/vim-tmux-navigator.tmux
@@ -3,11 +3,16 @@
 version_pat='s/^tmux[^0-9]*([.0-9]+).*/\1/p'
 
 is_vim="ps -o state= -o comm= -t '#{pane_tty}' \
-    | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?)(diff)?$'"
-tmux bind-key -n C-h if-shell "$is_vim" "send-keys C-h" "select-pane -L"
-tmux bind-key -n C-j if-shell "$is_vim" "send-keys C-j" "select-pane -D"
-tmux bind-key -n C-k if-shell "$is_vim" "send-keys C-k" "select-pane -U"
-tmux bind-key -n C-l if-shell "$is_vim" "send-keys C-l" "select-pane -R"
+  | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?|fzf)(diff)?$'"
+
+is_ssh="ps -o state= -o comm= -t '#{pane_tty}' \
+  | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?ssh$'"
+
+tmux bind-key -n C-h if-shell "$is_vim" "send-keys C-h" "if-shell \"$is_ssh\" 'send-keys C-b Left'  'select-pane -L'"
+tmux bind-key -n C-j if-shell "$is_vim" "send-keys C-j" "if-shell \"$is_ssh\" 'send-keys C-b Down'  'select-pane -D'"
+tmux bind-key -n C-k if-shell "$is_vim" "send-keys C-k" "if-shell \"$is_ssh\" 'send-keys C-b Up'    'select-pane -U'"
+tmux bind-key -n C-l if-shell "$is_vim" "send-keys C-l" "if-shell \"$is_ssh\" 'send-keys C-b Right' 'select-pane -R'"
+
 tmux_version="$(tmux -V | sed -En "$version_pat")"
 tmux setenv -g tmux_version "$tmux_version"
 
@@ -23,3 +28,6 @@ tmux bind-key -T copy-mode-vi C-j select-pane -D
 tmux bind-key -T copy-mode-vi C-k select-pane -U
 tmux bind-key -T copy-mode-vi C-l select-pane -R
 tmux bind-key -T copy-mode-vi C-\\ select-pane -l
+
+tmux bind-key -n C-q if-shell "$is_ssh" "send-keys C-l C-p Enter" "send-keys C-q"
+tmux bind-key -n M-l if-shell "$is_ssh" "send-keys 'ls -lA' Enter" "send-keys M-l"
