diff --git a/config/yazi/yazi.toml b/config/yazi/yazi.toml
index c42c96d..e959ec1 100644
--- a/config/yazi/yazi.toml
+++ b/config/yazi/yazi.toml
@@ -29,3 +29,11 @@ rules = [
   # { name = "*", use = [ "open", "reveal" ] },
   { mime = "*", use = [] },
 ]
+
+[plugin]
+# fetchers = []
+preloaders = []
+previewers = [
+  { name = "*/", run = "folder", sync = true },
+  { mime = "text/*", run = "code" },
+]
